src/webots/nodes/WbSolidReference.cpp

```diff
@@ -60,6 +60,8 @@ void WbSolidReference::updateName() {
   const bool linkToStaticEnvironment = name == STATIC_ENVIRONMENT;
   if (!linkToStaticEnvironment)
     mSolid = QPointer<WbSolid>(ts->findSolid(name, upperSolid()));
+  else
+    mSolid = NULL;
```

Comment:
This will likely cause a crash on the second line. Rather I would write something like:
```suggestion
    mSolid.clear();
```
However, I am not sure the usage of `QPointer` in this class. I don't see why we need a `QPointer`, maybe we could try to revert to a normal pointer instead...

Reply:
It do not crash thanks to the !linkToStaticEnvironment in the condition evaluated before will avoid evaluation of mSolid.isNull(). The NULL value is then nicely handle by the rest of webots, as far as my testing goes.
But I agree, clean() could be nicer, (i don't know anything about QPointer...).
May I lets you make the decision to keep it ?

Reply:
I studied it and I found that the `QPointer` is needed here (because the corresponding object may be deleted meanwhile and we don't want to have a dangling point here), so we should keep it. Also `mSolid = NULL;` is fully equivalent to `mSolid.clear();`, however I would prefer the second syntax as it shows that `mSolid` is not a standard pointer, but a `QPointer`.